api/tests/identity-access-management/unit/scripts/anonymize-users-in-batch.script.test.js

```diff
@@ -0,0 +1,88 @@
+import { AnonymizeUsersInBatchScript } from '../../../../src/identity-access-management/scripts/anonymize-users-in-batch.script.js';
+import { expect, sinon } from '../../../test-helper.js';
+
+describe('Unit | Identities Access Management | Scripts | Anonymize users in batch', function () {
+  let script;
+  let logger;
+  let anonymizeUser;
+
+  beforeEach(function () {
+    script = new AnonymizeUsersInBatchScript();
+    logger = {
+      info: sinon.spy(),
+      error: sinon.spy(),
+    };
+    anonymizeUser = sinon.stub();
+  });
+
+  describe('#handle', function () {
+    context('dryRun mode', function () {
+      it('should display the dryRun message', async function () {
+        // given
+        const file = [{ userId: 1 }, { userId: 2 }, { userId: 3 }];
+
+        // when
+        await script.handle({
+          options: {
+            file,
+            dryRun: true,
+            anonymizerId: 1234,
+          },
+          logger,
+          anonymizeUser,
+        });
+
+        // then
+        expect(logger.info).to.have.been.calledWith(`DryRun mode, 3 will be processed.`);
+      });
+    });
+
+    context('without dryRun mode', function () {
+      it('should sucessfull anonymized', async function () {
+        // given
+        const file = [{ userId: 1 }, { userId: 2 }, { userId: 3 }];
+
+        // when
+        await script.handle({
+          options: {
+            file,
+            dryRun: false,
+            anonymizerId: 1234,
+          },
+          logger,
+          anonymizeUser,
+        });
+
+        // then
+        expect(anonymizeUser.callCount).to.equal(3);
+        expect(logger.info).to.have.been.calledWith(`${file.length} processed successfull.`);
+      });
+
+      it('should return error if occured', async function () {
```

Comment:
Ne pas utiliser should mais plutôt le comportement attendu du point testé :

```suggestion
      context('when there is an error', async function () {
      it('stops the process and the rest of the file is not processed', async function () {
```

Reply:
Effectivement on ne mettra pas should, mais le script ne s'arrête pas si il y a une erreur.

Reply:
Ah bien reçu.